db/migrate/20170825144733_repopulate_first_published_at_from_change_history.rb

```diff
@@ -0,0 +1,39 @@
+class RepopulateFirstPublishedAtFromChangeHistory < ActiveRecord::Migration[5.1]
+  disable_ddl_transaction!
+
+  def up
+    min_date = '2016-02-29'
+    max_date = '2016-02-29T23:59:59'
```

Comment:
I'm not sure you have to be dealing with a 24 hour window here. I think it's all the same second.

```
[3] pry(main)> Edition.where("first_published_at > ? AND first_published_at < ?", "2016-02-29", "2016-02-29T23:59:59").count
   (1055.6ms)  SELECT COUNT(*) FROM "editions" WHERE (first_published_at > '2016-02-29' AND first_published_at < '2016-02-29T23:59:59')
=> 789244
```

```
[12] pry(main)> Edition.where("first_published_at > ? AND first_published_at <= ?", "2016-02-29T09:24:09", "2016-02-29T09:24:11").count
   (1056.2ms)  SELECT COUNT(*) FROM "editions" WHERE (first_published_at > '2016-02-29T09:24:09' AND first_published_at <= '2016-02-29T09:24:11')
=> 788272
```

You might know better than me at this point though whether the whole day is a problem or just that second.

Reply:
I think you're right, somewhere in analysing this I got the idea there were 2 different bulk update times that day, evidently not though:
```
SELECT COUNT(*) AS count_all, "editions"."first_published_at" AS editions_first_published_at FROM "editions" WHERE (first_published_at BETWEEN '2016-02-29' AND '2016-02-29T23:59:59') GROUP BY "editions"."first_published_at" HAVING (count(*) > 1000)
=> {Mon, 29 Feb 2016 09:24:10 UTC +00:00=>809051}
```